s3/src/main/scala/akka/stream/alpakka/s3/impl/HttpRequests.scala

```diff
@@ -11,12 +11,35 @@ import akka.http.scaladsl.model.{ContentTypes, RequestEntity, _}
 import akka.stream.alpakka.s3.S3Settings
 import akka.stream.scaladsl.Source
 import akka.util.ByteString
-
 import scala.collection.immutable.Seq
 import scala.concurrent.{ExecutionContext, Future}
 
 private[alpakka] object HttpRequests {
 
+  def listBucket(
+      bucket: String,
+      region: String,
+      prefix: Option[String] = None,
+      continuation_token: Option[String] = None
+  )(implicit conf: S3Settings): HttpRequest = {
+
+    val listTypeQuery: (String, String) = "list-type" -> "2"
```

Comment:
I guess this implementation is fine as is. 

It could probably be simplified by collecting all parameters as a `Seq[(String, Option[String])` and then converting that to a Map, filtering out the `None` values.

Something like

```scala
Seq("list-type" -> Some("2"),
  "prefix" -> prefix,
  "continuation-token" -> continuation_token.map(_.replace...)
).collect { case (k, Some(v)) => k -> v }.toMap
```